openspoor/visualisations/trackmap.py

```diff
@@ -79,82 +89,71 @@ def _add_luchtfoto(self) -> None:
-            except:
-                pass
+            if isinstance(item, folium.features.Choropleth):  # For linestrings
+                bboxes.extend(feature['bbox'][::-1] for feature in item.geojson.data['features'])
+            if isinstance(item, folium.features.GeoJson):  # For areas
+                bboxes.append([i for coords in item.get_bounds() for i in coords])
+            if isinstance(item, folium.map.Marker):  # For markers
+                bboxes.append(item.location * 2)  # Min and max bound are equal for points, hence the repeat
 
         if bboxes:  # Fit only if there are some items to show
             bounds = [min(map(lambda x: x[i], bboxes)) if i < 2 else max(map(lambda x: x[i], bboxes)) for i in range(4)]
             self.fit_bounds([bounds[:2], bounds[2:]])
 
-    def add(self, plotobject):
+    def add(self, plotobject) -> TrackMap:
         plotobject.add_to(self)
         return self
 
-    def show(self, makesmaller: bool = False):
+    def show(self, notebook: bool = False) -> Union[TrackMap, folium.Figure]:
         """
-        Fix the map so it is zoomed to a nice level and is displayed at a suitable size.
+        Show the map zoomed to a nice level and displayed at a suitable size.
 
-        :param makesmaller: A boolean for decreasing the size of the final plot. Designed for Databricks Notebooks
+        :param notebook: A boolean for decreasing the size of the final plot. Designed for use in Notebooks
         :return:
         """
         self._fix_zoom()
-        if makesmaller:
-            self.superfigure = folium.Figure(1000, 400)
-            self.add_to(self.superfigure)
-            return self.superfigure
+        if notebook:
+            figure = folium.Figure(1000, 400)
+            self.add_to(figure)
+            return figure
         else:
             return self
 
-    def save(self, save_name: Path):
+    def save(self, save_name: Path) -> None:
         """
         Save the map to a given directory
         :param save_name: A Path where the final .html file could be written to
-        :return:
+        :return: Nothing, the map is saved
         """
 
         self.show()
         super(TrackMap, self).save(save_name)
 
 
-class PlottingDataFrame(pd.DataFrame, PlotObject):
+class PlottingPoints(PlotObject):
     """
-    Add functionalities to a Pandas DataFrame so it can be plotted in a nice manner.
+    Add functionalities to a Pandas DataFrame, so it can be plotted in a nice manner.
     """
 
-    def __init__(self, df, lat_column: str = 'lat', lon_column: str = 'lon', popup: List[str] = None,
+    def __init__(self, data, popup: Optional[Union[str, List[str]]] = None,
+                 lat_column: str = 'lat', lon_column: str = 'lon',
                  colors: Dict[str, Dict[Tuple[float, float], str]] = None,
                  markertype: Optional[str] = None, marker_column: str = None, color_column: str = None,
```

Comment:
I guess not every string is possible here, since there is only a certain amount of marker options?

Reply:
That is correct. The docstring contains a link to the set of usable marker options. It is quite a list (that is regularly updated if I understand correctly), therefore I don't want to keep track of the possible options